homeassistant/components/amberelectric/manifest.json

```diff
@@ -6,5 +6,5 @@
   "documentation": "https://www.home-assistant.io/integrations/amberelectric",
   "iot_class": "cloud_polling",
   "loggers": ["amberelectric"],
-  "requirements": ["amberelectric==1.1.1"]
+  "requirements": ["amberelectric==2.0.0"]
```

Comment:
I see that you have a restriction on the pydantic version to not use 2.+ of pydantic. Please either make the library compatible with pydantic 2.+ or use another serializer/validator library eg mashumaro. We don't want to introduce new requirement conflicts for pydantic.

https://github.com/madpilot/amberelectric.py/compare/v1.1.1...v2.0.0#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R30

See: https://github.com/home-assistant/core/issues/99218

Reply:
Pydantic 2 wasn't compatible with home assistant, which is why I limited to it 1.x.

Has that changed?

Reply:
Libraries shouldn't restrict the version. Home Assistant currently restricts the version to not break unmigrated libraries that aren't compatible with pydantic 2.+.

Reply:
I'm not sure how to resolve that? The two versions are incompatible, and when I generated the library using a version 2 generator home assistant complained. So following the "we won't support version 2" I regenerated with version 1.

Is there is an easy way to support both versions?

Reply:
It's possible to support both. Check the issue for libraries that have migrated, or check the pydantic docs.